Add travis.yml

Krixt opened the pull request.

TheKrixt committed:
Add .travis.yml

The .travis.yml file was copied from the Travis CI Getting started guide
which can be found in this link:https://docs.travis-ci.com/user/getting-started/
or on the Travis CI homepage.

The file is added to initialize Travis CI.

Travis CI is a continuous integration service used to test and build
software. It enables open source projects to be tested.
https://en.wikipedia.org/wiki/Travis_CI
Currently the tests are failing due to a false configuration
in the -travis.yml file. This will be fixed in future commits.

TheKrixt committed:
Add a new version of ruby to the .travis.yml.

A new entry was created in the -travis.yml file to enable the
testing on ruby version 2.1.

TheKrixt committed:
WIP

paulmenzel commented:
@Krixt, what do we have to do to integrate with Travis CI? Get an account?

Additionally, the plan is to first add some working tests/specs which Travis CI then can run, right?

TheKrixt committed:
Updated the bin files as follows

Looks like your app's ./bin/rails is a stub that was generated by Bundler.

In Rails 4, your app's bin/ directory contains executables that are versioned
like any other source code, rather than stubs that are generated on demand.

Here's how to upgrade:

  bundle config --delete bin    # Turn off Bundler's stub generator
  rake rails:update:bin         # Use the new Rails 4 executables
  git add bin                   # Add bin/ to source control

You may need to remove bin/ from your .gitignore as well.

When you install a gem whose executable you want to use in your app,
generate it and add it to source control:

  bundle binstubs some-gem-name
  git add bin/new-executable

TheKrixt committed:
WIP

Update spec files.

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
WIP

Replace "rake" with "rails"

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
WIP

Database.yml

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
WIP

Remove: <%= app_secret %>

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
WIP

TheKrixt committed:
Add gem
Two gems are added to the Gemfile. This enable the use of capybara and
byebug.

TheKrixt committed:
Enable capybara

Exception for deprecated syntax was added.
Rspec no longer requires "expect" instead of "should".

TheKrixt committed:
Make byebug work

The line "require 'byebug'" wasd added to enable byebug
and to get rid of errors during tests.
Note:if you wish to use byebug you will have to uncomment the line.

TheKrixt committed:
WIP
